demos/projects/ESPRESSIF/aziotkit/components/sample-azure-iot/CMakeLists.txt

```diff
@@ -0,0 +1,33 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# SPDX-License-Identifier: MIT
+
+set(ROOT_PATH
+    ${CMAKE_CURRENT_LIST_DIR}/../../../../../..
+)
+
+set(COMPONENT_SOURCES

```

Comment:
This is new.
The new sample_azure_iot_pnp code has 3 files:
- sample_azure_iot_pnp_core.c, which has all the Azure IoT Client logic.
- sample_azure_iot_pnp.c, which has all the code that handles the PnP template data (implements the template).
- sample_azure_iot_pnp.h, which defines the interface for the two modules above to talk to each other.

In this aziotkit sample the code implements the functions of the sample_azure_iot_pnp.h interface, so it can plug in the ESP32 Azure IoT Kit board sensors and HIDs.

In this line the only c module included is the core, not including the sample_azure_iot_pnp.c.